src/classicalbacktrack.rs

```diff
@@ -694,9 +694,11 @@ impl<'a, Input: InputIndexer> MatchAttempter<'a, Input> {
                             data: *cg,
                         });
                         if Dir::FORWARD {
-                            cg.start = Some(pos)
+                            cg.start = Some(pos);
+                            cg.end = None;
```

Comment:
It is an important invariant that this is not necessary: if you enter a capture group its end must already be None. However this invariant was violated due to a bug!

I fixed the bug in 04a607e7b558919f2da3304a022536afb6ed1a0a so once again this ought not to be necessary. Please rebase on top of master, then replace this `cg.end = None` assignment with something like:

    debug_assert!(cg.end.is_none(), "Should not have already exited capture group we are entering")
    
likewise for `cg.start` below. Anyways great catch.